src/Knp/Snappy/AbstractGenerator.php

```diff
@@ -142,7 +125,9 @@ public function setOption($name, $value)
 
         $this->options[$name] = $value;
 
-        $this->logger->debug(\sprintf('Set option "%s".', $name), ['value' => $value]);
+        if (null !== $this->logger) {
```

Comment:
There is no need for these checks since in the constructor it is: `$this->logger = new NullLogger();`